src/services/lifi/api.ts

```diff
@@ -233,13 +237,26 @@ export class LiFiAPI {
       'Content-Type': 'application/json'
     }
 
-    // add the apiKey if specified only. Li Fi can function without an apiKey,
-    // it will just put a custom user rate limit
-    if (apiKey) {
-      this.#headers['x-lifi-api-key'] = apiKey
+    this.#apiKey = apiKey
+  }
+
+  setApiKeyIfAny() {
+    if (this.#apiKey) {
+      this.#headers['x-lifi-api-key'] = this.#apiKey
+      this.#apiKeyActivatedTimestamp = Date.now()
     }
   }
 
+  removeApiKey() {
+    if (!this.#apiKey || !this.#apiKeyActivatedTimestamp) return
+
+    const twoHoursPassed = Date.now() - this.#apiKeyActivatedTimestamp >= 120 * 60 * 1000
+    if (!twoHoursPassed) return
+
+    this.#apiKey = undefined
+    this.#apiKeyActivatedTimestamp = undefined
+  }
+
```

Comment:
This method doesn't remove the api key from `#headers`, meaning that if there is a rate limit once, the API key will remain until the user restarts their extension. Additionally, I don't see the point in resetting `this.#apiKey` as it will be missing for subsequent API failures. Imo we should never reset `this.#apiKey`, but instead remove the key from `this.#headers` and reset the timestamp

Reply:
should've been this.headers, it's a mistake